Firestore/Source/Local/FSTLRUGarbageCollector.mm

```diff
@@ -0,0 +1,101 @@
+#import "Firestore/Source/Local/FSTLRUGarbageCollector.h"
+#import <queue>
+
+#import "Firestore/Source/Local/FSTMutationQueue.h"
+#import "Firestore/Source/Local/FSTQueryCache.h"
+
+const FSTListenSequenceNumber kFSTListenSequenceNumberInvalid = -1;
+
+using std::priority_queue;
+
+class RollingSequenceNumberBuffer {
+ public:
+  RollingSequenceNumberBuffer(NSUInteger max_elements) : max_elements_(max_elements) {
+    queue_ = std::make_unique<priority_queue<FSTListenSequenceNumber> >();
+  }
+
+  RollingSequenceNumberBuffer(const RollingSequenceNumberBuffer &other) = delete;
```

Comment:
You only need to define one copy constructor and assignment operator as deleted to suppress the other.

You can prove it to yourself with a snippet like the following (though please delete after):

```c++
static_assert(!std::is_move_constructable<RollingSequenceNumberBuffer>::type, "no moves");
static_assert(!std::is_move_assignable<RollingSequenceNumberBuffer>::type, "no move assignments");
static_assert(!std::is_copy_constructable<RollingSequenceNumberBuffer>::type, "no copies");
static_assert(!std::is_copy_assignable<RollingSequenceNumberBuffer>::type, "no copy assignments");
```

https://stackoverflow.com/questions/33776697/deleting-copy-constructors-and-copy-assignment-operators-which-of-them-are-esse

Reply:
In the recommended fashion, I was just copying what I saw elsewhere. Any difference / preference for which one of each is kept?

For reference: https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/remote/datastore.h

Reply:
Do what I say, not what I do :-P.